lib/src/acl.rs

```diff
@@ -0,0 +1,386 @@
+            Self::Block => serializer.serialize_str(ACTION_BLOCK),
+            Self::Other(other) => serializer.serialize_str(format!("Other({})", other).as_str()),
+        }
+    }
+}
+
+#[test]
+fn prefix_is_match() {
+    let prefix = Prefix::new(Ipv4Addr::new(192, 168, 100, 0).into(), 16);
+
+    assert_eq!(
+        prefix.is_match(Ipv4Addr::new(192, 168, 100, 0).into()),
+        Some(16)
+    );
+    assert_eq!(
+        prefix.is_match(Ipv4Addr::new(192, 168, 200, 200).into()),
+        Some(16)
+    );
+
+    assert_eq!(prefix.is_match(Ipv4Addr::new(192, 167, 0, 0).into()), None);
+    assert_eq!(prefix.is_match(Ipv4Addr::new(192, 169, 0, 0).into()), None);
+
+    let prefix = Prefix::new(Ipv6Addr::new(0xFACE, 0, 0, 0, 0, 0, 0, 0).into(), 16);
+    assert_eq!(
+        prefix.is_match(Ipv6Addr::new(0xFACE, 1, 2, 3, 4, 5, 6, 7).into()),
+        Some(16)
+    );
+
+    let v4 = Ipv4Addr::new(192, 168, 200, 200);
+    let v4_as_v6 = v4.to_ipv6_mapped();
+
+    assert_eq!(Prefix::new(v4.into(), 8).is_match(v4_as_v6.into()), None);
+    assert_eq!(Prefix::new(v4_as_v6.into(), 8).is_match(v4.into()), None);
+}
+
+#[test]
+fn acl_lookup() {
+    let acl = Acl {
+        entries: vec![
+            Entry {
+                prefix: Prefix::new(Ipv4Addr::new(192, 168, 100, 0).into(), 16),
+                action: Action::Block,
+            },
+            Entry {
+                prefix: Prefix::new(Ipv4Addr::new(192, 168, 100, 0).into(), 24),
+                action: Action::Block,
+            },
+            Entry {
+                prefix: Prefix::new(Ipv4Addr::new(192, 168, 100, 0).into(), 8),
+                action: Action::Block,
+            },
+        ],
+    };
+
+    match acl.lookup(Ipv4Addr::new(192, 168, 100, 1).into()) {
+        Some(lookup_match) => {
+            assert_eq!(acl.entries[1], *lookup_match);
+        }
+        None => panic!("expected lookup match"),
+    };
+
+    match acl.lookup(Ipv4Addr::new(192, 168, 200, 1).into()) {
+        Some(lookup_match) => {
+            assert_eq!(acl.entries[0], *lookup_match);
+        }
+        None => panic!("expected lookup match"),
+    };
+
+    match acl.lookup(Ipv4Addr::new(192, 1, 1, 1).into()) {
+        Some(lookup_match) => {
+            assert_eq!(acl.entries[2], *lookup_match);
+        }
+        None => panic!("expected lookup match"),
+    };
+
+    if let Some(lookup_match) = acl.lookup(Ipv4Addr::new(1, 1, 1, 1).into()) {
+        panic!("expected no lookup match, got {:?}", lookup_match)
+    };
+}
+
+#[test]
+fn acl_json_parse() {
+    let input = r#"
+    { "entries": [
+        { "op": "create", "prefix": "1.2.3.0/24", "action": "BLOCK" },
+        { "op": "update", "prefix": "192.168.0.0/16", "action": "BLOCK" },
+        { "op": "create", "prefix": "23.23.23.23/32", "action": "ALLOW" },
+        { "op": "update", "prefix": "1.2.3.4/32", "action": "ALLOW" },
+        { "op": "update", "prefix": "1.2.3.4/8", "action": "ALLOW" }
```

Comment:
It looks like this testcase includes the `"op"` field, which is nice.  Is it just ignored then?  That's probably fine then!

Reply:
Yes, it's ignored (along with any other extraneous fields).

Thanks for pointing out the inconsistency with this test case and the integration test data. In [c18e3b6](https://github.com/fastly/Viceroy/pull/438/commits/c18e3b6e8234c0117b1e64ffc3078ab7079706e8), I've corrected that along with adding better comments to better document the JSON structure.